### PR TITLE
refactor: update module layout to new style

### PR DESCRIPTION
**Pull request by nathanielc**
This converts from <module_name>/mod.rs to simply <module_name>.rs. Additionally changes the tests to be integration tests so they consume the rust-ceramic-migrations-tests crate via its public interface. This means we can remove the allow_dead code directive as the tests directly consume the crate.

### CODE REVIEW

**Review by smrz2001 — APPROVED** [review]
🥳

**Comment by nathanielc** [comment]
Ok, so each module in the `test` directory becomes its own binary in addition to the unit test binary. The build process has been updated to collect all tests binaries and run them all as part of the docker image entrypoint.

@smrz2001 Would you mind reviewing again?

File: Dockerfile [hunk 1]
```diff
@@ -18,13 +18,13 @@ COPY . .
...
-    make $BUILD_MODE && \
-    cp ./target/release/rust-ceramic-migration-tests ./
+    make $BUILD_MODE
 
 FROM ubuntu:latest as tester
 
 COPY --from=builder /home/builder/rust-ceramic-migration-tests/env/.env /usr/bin/.env
-COPY --from=builder /home/builder/rust-ceramic-migration-tests/rust-ceramic-migration-tests /usr/bin
+COPY --from=builder /home/builder/rust-ceramic-migration-tests/test-binaries /test-binaries
+COPY --from=builder /home/builder/rust-ceramic-migration-tests/entrypoint.sh /usr/bin/rust-ceramic-migration-tests
```
Comment (smrz2001) [inline Dockerfile:27]: Should the target just be called `entrypoint.sh`? Will be clearer I think.

**Review by smrz2001 — APPROVED** [review]
LGTM apart from an editorial comment (not a must have just a suggestion).

File: Dockerfile (hunk 1 above)
Comment (nathanielc) [reply Dockerfile]: Agreed, updated.

**Pull request merged**